### PR TITLE
refactor fixconfig, rename to genjobs and move into config dir

### PR DESCRIPTION
**Pull request by BenTheElder**
this removes ~2700 lines of generated jobs from `prow/config.yaml`
Edit: it now also moves a lot of the kubernetes deployment specific jobs tests out of prow/ as well, and fixes those a bit

security repo jobs diff: https://www.diffchecker.com/yjeNhctE

### CODE REVIEW

**Comment by BenTheElder** [comment]
/hold
the security repo jobs test does not load the jobs dir, need to fix that first, I'll also move that over.

File: prow/config/jobtests/job_config_test.go [hunk 1]
```diff
@@ -489,179 +278,6 @@ func allJobs() ([]cfg.Presubmit, []cfg.Postsubmit, []cfg.Periodic, error) {
...
-			} else if current && pinned {
-				t.Errorf("%s: please remove from the pinnedJobs list", j)
-			} else if !current && v == "" {
-				t.Errorf("%s: pinning to a non-default version requires a non-empty reason for doing so", j)
-			}
-		}
-	}
-}
-
 func TestURLTemplate(t *testing.T) {
```
Comment (krzyzacy) [inline prow/config/jobtests/job_config_test.go]: I'd say those two tests are also k8s specific

File: config/jobs/kubernetes-security/generated-security-jobs.yaml [hunk 2]
```diff
@@ -0,0 +1,2678 @@
...
+      skip_report: true
+      spec:
+        containers:
+        - args:
+          - --ssh=/etc/ssh-security/ssh-security
+          - --repo=github.com/kubernetes-security/$(REPO_NAME)=$(PULL_REFS)
+          - --upload=gs://kubernetes-security-prow/pr-logs
+          - --timeout=75
+          - --
+          - --gcs-shared=gs://kubernetes-security-prow/bazel
```
Comment (krzyzacy) [inline config/jobs/kubernetes-security/generated-security-jobs.yaml]: we probably should also dump config.json flags here as well? (or when we migrate the actual presubmit jobs)

Comment (BenTheElder) [reply config/jobs/kubernetes-security/generated-security-jobs.yaml]: can I do that in a follow up? this PR is already XXL and I wanted to make sure there was no real diff between the old generation method and the new, I had to significantly refactor how it works internally.

Comment (BenTheElder) [reply config/jobs/kubernetes-security/generated-security-jobs.yaml]: https://www.diffchecker.com/yjeNhctE

File: prow/config/jobtests/job_config_test.go (hunk 1 above)
Comment (BenTheElder) [reply prow/config/jobtests/job_config_test.go]: ah yes, I'll move these

File: config/jobs/kubernetes-security/genjobs.go [hunk 3]
```diff
@@ -457,20 +489,22 @@ func main() {
...
-	// write the original bytes before the security repo section
-	_, err = f.Write(originalBytes[:securityRepoStart])
-	if err != nil {
-		log.Fatalf("Failed to write temp file: %v", err)
-	}
-	f.Sync()
-	io.WriteString(f, "\n")
+	// write the header
+	io.WriteString(f, "# Autogenerated by genjobs.go please do not edit!\n")
+	io.WriteString(f, "# To modify these jobs first update their kubernetes/kubernetes sources\n")
```
Comment (krzyzacy) [inline config/jobs/kubernetes-security/genjobs.go]: nit: kubernetes/test-infra sources?

File: prow/config/jobtests/job_config_test.go (hunk 1 above)
Comment (BenTheElder) [reply prow/config/jobtests/job_config_test.go]: done

**Comment by krzyzacy** [comment]
also cc @amwat we are moving the config test target :-)

File: prow/config/jobtests/job_config_test.go [hunk 4]
```diff
@@ -197,6 +191,7 @@ func TestRetestMatchJobsName(t *testing.T) {
 	}
 }
 
+// TODO(cjwagner): remove this when the submit-queue is removed
```
Comment (BenTheElder) [inline prow/config/jobtests/job_config_test.go]: FYI @cjwagner

File: config/jobs/kubernetes-security/genjobs.go (hunk 3 above)
Comment (BenTheElder) [reply config/jobs/kubernetes-security/genjobs.go]: can I also cover this in the same follow-up? will re-word.

**Comment by BenTheElder** [comment]
/retest
Directory not empty is back!?

**Comment by BenTheElder** [comment]
@krzyzacy tests are green now, PTAL

**Pull request merged**